### PR TITLE
Fix parsing ACPI tables address that share the same signature

### PR DESCRIPTION
**Pull request by raisfathin**

### CODE REVIEW

**Comment by raisfathin** [comment]
I have a doubt regarding these [methods](https://github.com/raisfathin/chipsec/blob/d334847007cc3863e1ac0a99b0fa2e501334acbf/source/tool/chipsec/hal/acpi.py#L377-L396).

To make it more consistent, I think we should make them return a list containing the results for all tables with the same signature instead of adding table index as an argument. The problem is that it is also used in other modules (for example: [1](https://github.com/raisfathin/chipsec/blob/master/source/tool/chipsec/hal/cpu.py#L102)).

What do you think?

File: source/tool/chipsec/hal/acpi.py [hunk 1]
```diff
@@ -328,6 +328,12 @@ def get_SDT( self, search_rsdp=True ):
         return (is_xsdt,sdt_pa,sdt,sdt_header)
 
 
+    def _add_to_tableList( self, signature, address ):
```
Comment (tweksteen) [inline source/tool/chipsec/hal/acpi.py]: What about using `defaultdict(list)`?

File: source/tool/chipsec/hal/acpi.py [hunk 2]
```diff
@@ -393,18 +399,20 @@ def get_ACPI_table( self, name, isfile = False ):
...
-        logger().log( "==================================================================" )
-        # print table header
-        logger().log( table_header )
-        print_buffer( table_header_blob )
-        # print table contents
-        logger().log( '' )
-        logger().log( table )
-        print_buffer( table_blob )
-        logger().log( '' )
+        for i in xrange(len(self.tableList[name])):
```
Comment (tweksteen) [inline source/tool/chipsec/hal/acpi.py]: `for i, addr in enumerate(self.tableList[name]):`

**Comment by raisfathin** [comment]
Done! Ready for another review @tweksteen

**Comment by tweksteen** [comment]
LGTM

File: source/tool/chipsec/hal/acpi.py [hunk 3]
```diff
@@ -363,48 +364,58 @@ def print_ACPI_table_list(self):
...
-        table_header_blob  = None
-        table_blob = None
-        t_data = None
-        if isfile == True:
-            t_data = chipsec.file.read_file( name )
-        else:
-            t_size = self.cs.mem.read_physical_mem_dword( self.tableList[name] + 4 )
-            t_data = self.cs.mem.read_physical_mem( self.tableList[name], t_size )
+        acpi_tables_data = []
+        if isfile == True:
```
Comment (tweksteen) [inline source/tool/chipsec/hal/acpi.py]: `if isfile:`

Comment (raisfathin) [inline source/tool/chipsec/hal/acpi.py]: Done.

**Pull request merged**